lib/overcommit/hook/pre_commit/java_checkstyle.rb

```diff
@@ -3,18 +3,25 @@ module Overcommit::Hook::PreCommit
   #
   # @see http://checkstyle.sourceforge.net/
   class JavaCheckstyle < Base
-    MESSAGE_REGEX = /^(\[[^\]]+\]\s+)?(?<file>(?:\w:)?[^:]+):(?<line>\d+)/
+    MESSAGE_REGEX = /^(?<type>\[[^\]]+\]\s+)?(?<file>(?:\w:)?[^:]+):(?<line>\d+)/
+
+    MESSAGE_TYPE_CATEGORIZER = lambda do |type|
+      # Type may be nil if checkstyle doesn't output a 'tag'
+      type ||= ''
+
+      type.include?('WARN') || type.include?('INFO') ? :warning : :error
```

Comment:
Actually, I wasn't trying to handle `WARNING` or `INFORMATION`.  I'll change it to what you suggested.  Looks _much_ better.

Regarding `INFO` messages, I took a bit of a shortcut and figured we could just report both `INFO` and `WARN` messages as warnings.